sidecar/proxy/envoy/manager.go

```diff
@@ -485,6 +484,12 @@ func sanitizeRules(ruleList []api.Rule) {
 					}
 				}
 			}
+
+			sum = 0
+			for j := range rule.Route.Backends {
+				sum += rule.Route.Backends[j].Weight
+				rule.Route.Backends[j].Weight = sum
+			}
```

Comment:
@rshriram what is the benefit of using [Approach A](https://lyft.github.io/envoy/docs/configuration/http_conn_man/route_config/traffic_splitting.html#a-weight-based-cluster-selection) over [Approach B](https://lyft.github.io/envoy/docs/configuration/http_conn_man/route_config/traffic_splitting.html#b-probabilistic-route-selection)?  Right now this PR is using Approach B